### PR TITLE
Re-org/updated step library and added additional steps

### PR DESCRIPTION
**Pull request by nikokaoja**
### Changed

- Re-org and re-name step library
- Update workflows according to new step library org
### Added

- `OntologyFromRules` step to step library
- `SHACLFromRules` step to step library
- `DownloadTransformationRulesFromGitHub` to step library

### Improved
- `data_model_generation` workflow has been extended to produce ontological and shape constraints representation
- input parameters description for workflow steps in step library

### CODE REVIEW

File: cognite/neat/workflows/steps/step_model.py [hunk 1]
```diff
@@ -7,6 +8,16 @@
...
 
 
+class StepCategory(StrEnum):
+    RulesImporter = "rules | importer"
+    RulesParser = "rules | parser"
+    RulesExporter = "rules | exporter"
+    GraphExtractor = "graph | extractor"
+    GraphTransformer = "graph | transformer"
+    GraphStore = "graph | store"
+    GraphLoader = "graph | loader"
```
Comment (doctrino) [inline cognite/neat/workflows/steps/step_model.py:18]: Can we simply use the module name for this grouping instead? That will force all steps to be put into the correct module.

File: docs/api/workflows/steps/graph.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+::: cognite.neat.workflows.steps.lib.graph_store
+::: cognite.neat.workflows.steps.lib.graph_extractor
+::: cognite.neat.workflows.steps.lib.graph_loader
+::: cognite.neat.workflows.steps.lib.graph_transformer
```
Comment (doctrino) [inline docs/api/workflows/steps/graph.md:4]: Is it worth having some extra headers inbetween the modules to make navigation easier.

File: cognite/neat/workflows/steps/lib/rules_exporter.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+__all__ = ["GraphQLSchemaFromRules", "OntologyFromRules", "SHACLFromRules", "GraphCaptureSpreadsheetFromRules"]
+
+
+class GraphQLSchemaFromRules(Step):
+    """
+    This step generates GraphQL schema from data model defined in transformation rules
+    """
+
+    description = "This step generates GraphQL schema from data model defined in transformation rules."
+    category = StepCategory.RulesExporter
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/workflows/steps/lib/rules_exporter.py]: Solved using module name

File: cognite/neat/workflows/steps/step_model.py (hunk 1 above)
Comment (nikokaoja) [reply cognite/neat/workflows/steps/step_model.py]: removed

**Pull request merged**